src/vivarium_gates_nutrition_optimization_child/components/wasting.py

```diff
@@ -0,0 +1,857 @@
+)
+from vivarium_public_health.utilities import EntityString
+
+from vivarium_gates_nutrition_optimization_child.constants import (
+    data_keys,
+    data_values,
+    metadata,
+    models,
+    scenarios,
+)
+from vivarium_gates_nutrition_optimization_child.constants.data_keys import WASTING
+from vivarium_gates_nutrition_optimization_child.utilities import get_random_variable
+
+
+class ChildWasting:
+    def __init__(self):
+        self.dynamic_model = DynamicChildWasting()
+        self.static_model = StaticChildWasting()
+
+    @property
+    def sub_components(self):
+        return [
+            self.dynamic_model,
+            self.static_model,
+        ]
+        # noinspection PyAttributeOutsideInit
+
+    @property
+    def name(self):
+        return f"child_wasting"
+
+    def __repr__(self):
+        return "ChildWasting()"
+
+    def setup(self, builder: Builder):
+        self.population_view = builder.population.get_view(
+            [
+                "alive",
+                "age",
+                "sex",
+                self.dynamic_model.state_column,
+                self.static_model.propensity_column_name,
+            ]
+        )
+        self.exposure = builder.value.register_value_producer(
+            f"{self.name}.exposure",
+            source=self.get_current_exposure,
+            requires_columns=[self.static_model.propensity_column_name],
+            requires_values=[self.static_model.exposure_pipeline_name],
+            preferred_post_processor=get_exposure_post_processor(
+                builder, EntityString(f"risk_factor.{self.name}")
+            ),
+        )
+
+    def get_current_exposure(self, index: pd.Index) -> pd.Series:
+        pop = self.population_view.get(index)
+        exposure = pop[self.dynamic_model.state_column].apply(models.get_risk_category)
+        under_six_months = (pop["age"] < data_values.WASTING.START_AGE) & (
+            pop["alive"] == "alive"
+        )
+        if under_six_months.any():
+            exposure[under_six_months] = self.static_model.exposure(
+                pop[under_six_months].index
+            )
+        return exposure
+
+
+class StaticChildWasting(Risk):
+    def __init__(self):
+        """
+        Parameters
+        ----------
+        risk :
+            the type and name of a risk, specified as "type.name". Type is singular.
+        """
+        self.risk = EntityString("risk_factor.child_wasting")
+        name = "static_child_wasting"
+        self.configuration_defaults = self._get_configuration_defaults()
+        self.exposure_distribution = self._get_exposure_distribution()
+        self._sub_components = [self.exposure_distribution]
+
+        self._randomness_stream_name = f"initial_{name}_propensity"
+        self.propensity_column_name = f"{name}_propensity"
+        self.propensity_pipeline_name = f"{name}.propensity"
+        self.exposure_pipeline_name = f"{name}.exposure"
+
+
+class WastingTreatment(Risk):
```

Comment:
the reason for this subclass is to pass in the propensities from the static model